Build failure due to incompatible-pointer-types: `_Complex double *` vs `double (*)[2]`

On EndeavourOS, where

- GCC 14.2.1
- python 3.12.5
- fftw 3.3.10

installation fails (with #395 applied):

```console
$ pip install .
…
      INFO:root:gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -ffat-lto-objects -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -march=x86-64 -mtune=generic -O3 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -g -ffile-prefix-map=/build/python/src=/usr/src/debug/python -flto=auto -fPIC -DPYFFTW_HAVE_DOUBLE=1 -DPYFFTW_HAVE_DOUBLE_OMP=1 -DPYFFTW_HAVE_DOUBLE_THREADS=0 -DPYFFTW_HAVE_DOUBLE_MULTITHREADING=1 -DPYFFTW_HAVE_DOUBLE_MPI=0 -DPYFFTW_HAVE_SINGLE=1 -DPYFFTW_HAVE_SINGLE_OMP=1 -DPYFFTW_HAVE_SINGLE_THREADS=0 -DPYFFTW_HAVE_SINGLE_MULTITHREADING=1 -DPYFFTW_HAVE_SINGLE_MPI=0 -DPYFFTW_HAVE_LONG=1 -DPYFFTW_HAVE_LONG_OMP=1 -DPYFFTW_HAVE_LONG_THREADS=0 -DPYFFTW_HAVE_LONG_MULTITHREADING=1 -DPYFFTW_HAVE_LONG_MPI=0 -DPYFFTW_HAVE_MPI=0 -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/tmp/pip-req-build-qsk682eo/include -I/tmp/pip-req-build-qsk682eo/pyfftw -I/tmp/pip-build-env-298ogx93/overlay/lib/python3.12/site-packages/numpy/_core/include -I/tmp/tmp.1NBemsk2Au/include -I/usr/include -I/tmp/tmp.1NBemsk2Au/include -I/usr/include/python3.12 -I/tmp/tmp.1NBemsk2Au/include -I/usr/include/python3.12 -c pyfftw/pyfftw.c -o build/temp.linux-x86_64-cpython-312/pyfftw/pyfftw.o
      pyfftw/pyfftw.c: In function ‘__pyx_f_6pyfftw_6pyfftw__fftw_plan_guru_dft’:
      pyfftw/pyfftw.c:13647:114: error: passing argument 5 of ‘fftw_plan_guru_dft’ from incompatible pointer type [-Wincompatible-pointer-types]
      13647 |   __pyx_r = ((void *)fftw_plan_guru_dft(__pyx_v_rank, __pyx_v_dims, __pyx_v_howmany_rank, __pyx_v_howmany_dims, ((cdouble *)__pyx_v__in), ((cdouble *)__pyx_v__out), (__pyx_v_direction[0]), __pyx_v_flags));
            |                                                                                                                 ~^~~~~~~~~~~~~~~~~~~~~~~
            |                                                                                                                  |
            |                                                                                                                  double (*)[2]
      In file included from pyfftw/pyfftw.c:1335:
      /usr/include/fftw3.h:465:1: note: expected ‘_Complex double *’ but argument is of type ‘double (*)[2]’
        465 | FFTW_DEFINE_API(FFTW_MANGLE_DOUBLE, double, fftw_complex)
            | ^
      pyfftw/pyfftw.c:13647:140: error: passing argument 6 of ‘fftw_plan_guru_dft’ from incompatible pointer type [-Wincompatible-pointer-types]
      13647 |   __pyx_r = ((void *)fftw_plan_guru_dft(__pyx_v_rank, __pyx_v_dims, __pyx_v_howmany_rank, __pyx_v_howmany_dims, ((cdouble *)__pyx_v__in), ((cdouble *)__pyx_v__out), (__pyx_v_direction[0]), __pyx_v_flags));
            |                                                                                                                                           ~^~~~~~~~~~~~~~~~~~~~~~~~
            |                                                                                                                                            |
            |                                                                                                                                            double (*)[2]
      /usr/include/fftw3.h:465:1: note: expected ‘_Complex double *’ but argument is of type ‘double (*)[2]’
        465 | FFTW_DEFINE_API(FFTW_MANGLE_DOUBLE, double, fftw_complex)
            | ^
```

In pyFFTW the complex floating-point numbers are defined as two-element arrays:

https://github.com/pyFFTW/pyFFTW/blob/89e2d2c208b7be6298625ab19edb9f9cc0d31303/pyfftw/pyfftw.pxd#L80-L82

GCC 11.4.1 on AlmaLinux 9.4 issues the warning that they are incompatible, but exits with zero.

--- a/setup.py
+++ b/setup.py
@@ -647,9 +647,7 @@ class custom_build_ext(build_ext):
         ]
         # to avoid a meaningless warning
         self._pyfftw_define_macros.append(("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION"))
-        if get_platform() in ('win32', 'win-amd64'):
-            # because Numpy 2.0 includes complex.h
-            self._pyfftw_define_macros.append(("FFTW_NO_Complex", "1"))
+        self._pyfftw_define_macros.append(("FFTW_NO_Complex", 1))
 
         # call `extend()` to keep argument set neither by sniffer nor by
         # user. On windows there are includes set automatically, we

